tidy3d/components/spice/analysis/dc.py

```diff
@@ -0,0 +1,75 @@
+"""
+This class defines standard SPICE electrical_analysis types (electrical simulations configurations).
+"""
+
+from typing import Optional, Tuple, Union
+
+import pydantic.v1 as pd
+
+from tidy3d.components.base import Tidy3dBaseModel
+from tidy3d.components.spice.sources.types import VoltageSourceType
+from tidy3d.components.tcad.types import HeatChargeMonitorTypes
+from tidy3d.components.types import annotate_type
+
+
+class ChargeToleranceSpec(Tidy3dBaseModel):
+    """
+    This class sets some Charge tolerance parameters.
+
+    Example
+    -------
+    >>> import tidy3d as td
+    >>> charge_settings = td.ChargeToleranceSpec(abs_tol=1e8, rel_tol=1e-10, max_iters=30)"""
+
+    abs_tol: pd.PositiveFloat = pd.Field(
+        default=1e10,
+        title="Absolute tolerance.",
+        description="Absolute tolerance used as stop criteria when converging towards a solution. Should be "
+        "equivalent to the"
+        "SPICE ABSTOL DC transfer parameter. TODO MARC units, TODO check equivalence. TODO what does this mean?",
+    )
+
+    rel_tol: pd.PositiveFloat = pd.Field(
+        default=1e-10,
+        title="Relative tolerance.",
+        description="Relative tolerance used as stop criteria when converging towards a solution.  Should be equivalent to the"
+        "SPICE RELTOL DC transfer parameter. TODO MARC units, TODO check equivalence. TODO what does this "
+        "mean?",
+    )
+
+    max_iters: pd.PositiveInt = pd.Field(
+        default=30,
+        title="Maximum number of iterations.",
+        description="Indicates the maximum number of iterations to be run. "
+        "The solver will stop either when this maximum of iterations is met "
+        "or when the tolerance criteria has been met. Should be equivalent to the ngspice ITL1 parameter. TODO units, "
+        "TODO devsim ngspice equivalence",
+    )
+
+
+class SteadyDCAnalysis(Tidy3dBaseModel):
```

Comment:
@daquinteroflex After some thought and reading through the ngspice manual I decided change this to `SteadyDCAnalysis` (from `TransferFunctionDC`). The main reason for this is, that while reading the ngspice manual it appeared that "transfer" is mostly used in relation to output whereas they're still calling it a DC analysis when talking about analysis type. So my suggestion would be, if you feel like changing it, to change the data array names that we have for the I-V and C-V curves. These could be called something like CapacitanceDCTransferFunction or something like that.

As always, we can discuss further with the broader team, including @dbochkov-flexcompute @momchil-flex and prash